npm/src/lib/libDef.js

```diff
@@ -186,13 +189,23 @@ export async function getLocalLibDefFlowVersions(
         }
       }
 
-      let [_1, _2, range, major, minor, patch] = matches;
+      let [
+        _1, _2, range, major, minor, patch,
+        _3, upRange, upMajor, upMinor, upPatch
+      ] = matches;
       range = _validateVersionRange(range, itemPath, validationErrors);
       major = _validateVersionNumPart(major, "major", itemPath, validationErrors);
       minor = _validateVersionPart(minor, "minor", itemPath, validationErrors);
       patch = _validateVersionPart(patch, "patch", itemPath, validationErrors);
 
-      const flowVersion = {range, major, minor, patch};
+      if (upMajor) {
+        upRange = _validateVersionRange(upRange, itemPath, validationErrors);
+        upMajor = _validateVersionNumPart(upMajor, "major", itemPath, validationErrors);
+        upMinor = _validateVersionPart(upMinor, "minor", itemPath, validationErrors);
+        upPatch = _validateVersionPart(upPatch, "patch", itemPath, validationErrors);
+      }
+
```

Comment:
We should probably also validate that the range makes sense (i.e. something like `flow->=v0.23.x_<=v0.23.x` doesn't make much sense).

(Feel free to `// TODO: ` this, though -- probably not critical right now)